src/components/checkbox/index.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { Icon } from '../index';
+
+const Checkbox = (props) => (
+  <label
+    className={ cx('checkbox-wrapper', {
+      'checkbox-wrapper--checked': props.checked
+    }) }
+    onClick={ (e) => {
```

Comment:
If an input component like a text input or a checkbox is wrapped inside a label tag, the input component automatically gets focused when you're clicking anything inside the label tag even if it's not the input element itself. So in case of a checkbox, clicking inside the label makes the checkbox trigger a change event. On the other hand, it would be useful if we pass if the checkbox is checked or not to the event listener.

You can remove the onClick from the label tag and set an onChange event listener on the input tag:

```
<input type=... checked=... onChange={ () => this.props.onChange(e.target.checked) } />
```

Reply:
Ahh since I realized that you wanted it to be fully controlled. Hmm, good question. Let's go with a fully controlled component so in this case the "checked" property should not be passed to the event listener. Whatdyathink?

Reply:
I was thinking about this. I figured we could get rid of the `<input />` as it is, and use something like this instead:
```jsx
<div role="checkbox" aria-checked={ this.state.checked } /> 
```
This way, we don't need to worry about anything related to `<label>` and `onChange`. Even the browser is going to assume that it's handling with a checkbox.

What do you think @ruffle1986 ?

Reply:
Alrite. Sounds good. Can't wait to see the implementation.